### PR TITLE
Only require contexts when protection is on

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @fejta

### CODE REVIEW

File: prow/config/branch_protection.go [hunk 1]
```diff
@@ -250,9 +250,10 @@ func (c *Config) GetPolicy(org, repo, branch string, b Branch) (*Policy, error)
 
 	// Automatically require contexts from prow which must always be present
 	if prowContexts, _, _ := BranchRequirements(org, repo, branch, c.Presubmits); len(prowContexts) > 0 {
-		// Error if protection is disabled
+		shouldRequireContexts := c.BranchProtection.ProtectTested
```
Comment (fejta) [inline prow/config/branch_protection.go]: Can we get a unit test for this?

**Comment by stevekuznetsov** [comment]
@fejta changed slightly to entirely opt out of protection instead of opting into non-protection, added a test

PTAL

**Pull request merged**